tests/k8s/test_k8s.py

```diff
@@ -256,566 +142,278 @@ def k8s_flow_with_needs(docker_images):
             )
-    try:
-        args = set_deployment_parser().parse_args(['--uses', 'DocReplaceExecutor'])
-        with Deployment(args) as external_deployment:
-            ports = [args.port for args in external_deployment.pod_args['pods'][0]]
-            flow = Flow(name='k8s_flow-with_external_deployment', port=9090).add(
-                name='external_executor',
-                external=True,
-                host=f'172.17.0.1',
-                port=ports[0],
-            )
-
-            namespace = 'test-flow-with-external-deployment'
-            dump_path = os.path.join(str(tmpdir), namespace)
-            flow.to_kubernetes_yaml(dump_path, k8s_namespace=namespace)
-
-            from kubernetes import client
-
-            api_client = client.ApiClient()
-            core_client = client.CoreV1Api(api_client=api_client)
-            app_client = client.AppsV1Api(api_client=api_client)
-            await create_all_flow_deployments_and_wait_ready(
-                dump_path,
-                namespace=namespace,
-                api_client=api_client,
-                app_client=app_client,
-                core_client=core_client,
-                deployment_replicas_expected={
-                    'gateway': 1,
-                },
-                logger=logger,
-            )
-            resp = await run_test(
-                flow=flow,
-                namespace=namespace,
-                core_client=core_client,
-                endpoint='/',
-            )
-        docs = resp[0].docs
-        assert len(docs) == 100
-        for doc in docs:
-            assert doc.text == 'executor was here'
-        core_client.delete_namespace(namespace)
-    except Exception as exc:
-        logger.error(f' Exception raised {exc}')
-        raise exc
-
-
-@pytest.mark.asyncio
-@pytest.mark.timeout(3600)
-@pytest.mark.parametrize(
-    'docker_images',
-    [['test-executor', 'jinaai/jina']],
-    indirect=True,
-)
-async def test_flow_with_external_k8s_deployment(logger, docker_images, tmpdir):
-    try:
-        namespace = 'test-flow-with-external-k8s-deployment'
-        from kubernetes import client
-
-        api_client = client.ApiClient()
-        core_client = client.CoreV1Api(api_client=api_client)
-        app_client = client.AppsV1Api(api_client=api_client)
-
-        await _create_external_deployment(api_client, app_client, docker_images, tmpdir)
 
+    args = set_deployment_parser().parse_args(['--uses', 'DocReplaceExecutor'])
+    with Deployment(args) as external_deployment:
+        ports = [args.port for args in external_deployment.pod_args['pods'][0]]
         flow = Flow(name='k8s_flow-with_external_deployment', port=9090).add(
             name='external_executor',
             external=True,
-            host='external-deployment.external-deployment-ns.svc',
-            port=GrpcConnectionPool.K8S_PORT,
+            host=f'172.17.0.1',
+            port=ports[0],
```

Comment:
??

Reply:
Oh it's because the diff is matching the wrong place. It matched the duplicate test_flow_with_external_k8s_deployment